### PR TITLE
Added workflow level matchable attribute get for execution config

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
Signed-off-by: Prafulla Mahindrakar <prafulla.mahindrakar@gmail.com>

# TL;DR

### For no workflow level matchable attribute

* At project domain level
```
flytectl get workflow-execution-config -p flytesnacks -d development                               
{"project":"flytesnacks","domain":"development","max_parallelism":10,"security_context":{"run_as":{"k8s_service_account":"default"}},"raw_output_data_config":{"output_location_prefix":"cliOutputLocationPrefix"},"labels":{"values":{"cliLabelKey":"cliLabelValue"}},"annotations":{"values":{"cliAnnotationKey":"cliAnnotationValue"}}}
```

* For non existent workflow level matchable attribute it retrieves project-domain level attribute if it exists
 or else it would return empty
```
 flytectl get workflow-execution-config -p flytesnacks -d development core.control_flow.run_conditions.multiplier
{"project":"flytesnacks","domain":"development","workflow":"core.control_flow.run_conditions.multiplier","max_parallelism":10,"security_context":{"run_as":{"k8s_service_account":"default"}},"raw_output_data_config":{"output_location_prefix":"cliOutputLocationPrefix"},"labels":{"values":{"cliLabelKey":"cliLabelValue"}},"annotations":{"values":{"cliAnnotationKey":"cliAnnotationValue"}}}
```

```
metadata:
  annotations:
    cliAnnotationKey: cliAnnotationValue
    cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
  creationTimestamp: "2022-03-30T13:21:43Z"
  labels:
    cliLabelKey: cliLabelValue
    domain: development
    execution-id: f5da971b6ff0a4ff591d
```

### For workflow level matchable attribute

```
flytectl get workflow-execution-config -p flytesnacks -d development  core.flyte_basics.lp.go_greet
{"project":"flytesnacks","domain":"development","workflow":"core.flyte_basics.lp.go_greet","max_parallelism":10,"security_context":{"run_as":{"k8s_service_account":"default"}},"raw_output_data_config":{"output_location_prefix":"cliOutputLocationPrefix"},"labels":{"values":{"workflowLevelKey":"workflowLevelValue"}},"annotations":{"values":{"workflowLevelKey":"workflowLevelValue"}}}
```

```
metadata:
  annotations:
    workflowLevelKey: workflowLevelValue
    cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
  creationTimestamp: "2022-03-30T13:20:29Z"
  labels:
    workflowLevelKey: workflowLevelValue
```

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [X] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
 https://github.com/flyteorg/flyte/issues/2287

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/execution_manager.go [hunk 1]
```diff
@@ -490,15 +490,19 @@ func (m *ExecutionManager) getExecutionConfig(ctx context.Context, request *admi
...
 			return workflowExecConfig, nil
 		}
+		if launchPlan.Spec.WorkflowId != nil {
+			workflowName = launchPlan.Spec.WorkflowId.Name
+		}
 	}
 
 	matchableResource, err := util.GetMatchableResource(ctx, m.resourceManager,
-		admin.MatchableResource_WORKFLOW_EXECUTION_CONFIG, request.Project, request.Domain)
+		admin.MatchableResource_WORKFLOW_EXECUTION_CONFIG, request.Project, request.Domain, workflowName)
```
Comment (katrogan) [inline pkg/manager/impl/execution_manager.go:505]: should we also pass the launch plan name here? https://github.com/flyteorg/flyteadmin/blob/master/pkg/manager/interfaces/resource.go#L35

File: pkg/manager/impl/util/shared.go [hunk 2]
```diff
@@ -235,13 +235,14 @@ func GetTaskExecutionModel(
 	return &taskExecutionModel, nil
 }
 
-// GetMatchableResource gets matchable resource for resourceType and project - domain combination.
+// GetMatchableResource gets matchable resource for resourceType and project - domain - workflow combination.
 // Returns nil with nothing is found or return an error
 func GetMatchableResource(ctx context.Context, resourceManager interfaces.ResourceInterface, resourceType admin.MatchableResource,
-	project, domain string) (*interfaces.ResourceResponse, error) {
+	project, domain, name string) (*interfaces.ResourceResponse, error) {
```
Comment (katrogan) [inline pkg/manager/impl/util/shared.go:241]:
```suggestion
	project, domain, workflowName string) (*interfaces.ResourceResponse, error) {
```

File: pkg/manager/impl/execution_manager.go (hunk 1 above)
Comment (katrogan) [reply pkg/manager/impl/execution_manager.go:505]: I see it's not used, feel free to ignore this comment as long as we capture this in an issue!

**Comment by pmahindrakar-oss** [comment]
@katrogan Elaborated more the non existent workflow attribute fetch behavior in the description along with the example

```
 For non existent workflow level matchable attribute it retrieves project-domain level attribute if it exists
 or else it would return empty
```

**Review by katrogan — APPROVED** [review]
